PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/appservice/SSHTerminalManager.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.serviceexplorer.azure.appservice;
+
+import com.microsoft.azuretools.utils.AzureCliUtils;
+import com.microsoft.azuretools.utils.CommandUtils;
+import com.microsoft.intellij.util.PatternUtils;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.plugins.terminal.ShellTerminalWidget;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+public enum SSHTerminalManager {
+    INSTANCE;
+
+    private static final Logger logger = Logger.getLogger(SSHTerminalManager.class.getName());
+
+    private static final String SSH_INTO_WEB_APP_ERROR_DIALOG_TITLE = "SSH into Web App Error";
+    private static final String SSH_INTO_WEB_APP_ERROR_MESSAGE = "Failed to SSH into Web App. Please try again.";
+    private static final String SSH_INTO_WEB_APP_ERROR_MESSAGE_WITH_NAME = "Failed to SSH into Web App (%s). Please try again.";
+    private static final String CLI_DIALOG_MESSAGE_DEFAULT =
+            "Failed to invoke Azure CLI. Please make sure Azure CLI is installed and signed-in with same account.";
+    private static final String SSH_INTO_WEB_APP_ERROR_MESSAGE_NOT_SUPPORT_WINDOWS = "Azure SSH is only supported for Linux web apps";
+    private static final String SSH_INTO_WEB_APP_ERROR_MESSAGE_NOT_SUPPORT_DOCKER = "Azure SSH is only supported for Linux web apps (Not in custom container).";
+    private static final String OS_LINUX = "linux";
+    private static final String WEB_APP_FX_VERSION_PREFIX = "DOCKER|";
+    private static final String CMD_SSH_TO_LOCAL_PROXY =
+            "ssh -o StrictHostKeyChecking=no -o \"UserKnownHostsFile /dev/null\" -o \"LogLevel ERROR\" root@127.0.0.1 -p %s";
+    private static final String CMD_SSH_TO_LOCAL_PWD = "Docker!";
+
+    /**
+     * these actions (validation, etc) before ssh into web app.
+     *
+     * @param subscriptionId
+     * @param os
+     * @param fxVersion
+     * @return
+     */
+    public boolean beforeExecuteAzCreateRemoteConnection(String subscriptionId, String os, String fxVersion) {
+        try {
+            // check to confirm that azure cli is installed.
+            if (!AzureCliUtils.isCliInstalled()) {
+                DefaultLoader.getUIHelper().showError(CLI_DIALOG_MESSAGE_DEFAULT, SSH_INTO_WEB_APP_ERROR_DIALOG_TITLE);
```

Comment:
As all the validation share the same error message, shall we just check the subscription directly? Which will cover the validation in above steps

Reply:
yes. it should be. updated.